### PR TITLE
Added st nodes to layout

### PR DESCRIPTION
**Pull request by hkumarmk**

### CODE REVIEW

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by bodepd** [comment]
this build actually failed (if you log in, you see 3 and not 4 osds). You need to add some kind of validation check to our osd code so that we can validate this role (we should also validate osd creation for the stmon role as well).

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by amar266** [comment]
please test this

**Comment by bodepd** [comment]
I believe this was caused by the Github failures. If you guys already figured this out, please remember to root cause build failures so that I don't have to perform the analysis.

**Comment by bodepd** [comment]
please test this

File: manifests/ceph/osd.pp [hunk 1]
```diff
@@ -85,6 +85,13 @@
   }
 
   ##
+  # Ceph osd validation check
+  ##
+  rjil::test::check { 'ceph-osd':
```
Comment (bodepd) [inline manifests/ceph/osd.pp]: this never gets run b/c the test::checks are only run by consul and not as validation checks, we probably need to make this more clear.

Also, I can see that for the last build this check passes even though this osd is not actually registered with the ceph mons

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
last test failed because of some kind of data corruption in cassandra.

This is the error I see from contrail-api logs
self = <vnc_cfg_api_server.vnc_cfg_ifmap.VncCassandraClient object>
self._create_prop = <bound method VncCassandraClient._create_prop of..._server.vnc_cfg_ifmap.VncCassandraClient object>>
bch = <pycassa.batch.CfMutator object>
obj_ids = 'FQ Name: /fq-name-to-uuid/global_system_config:d...ady with ID: 7c9a4346-e205-4f32-bac5-3cec1daf3de6'
field = 64512
<type 'exceptions.TypeError'>: string indices must be integers, not str
    __class__ = <type 'exceptions.TypeError'>
    __delattr__ = <method-wrapper '__delattr__' of exceptions.TypeError object>
    __dict__ = {}
    __doc__ = 'Inappropriate argument type.'
    __format__ = <built-in method __format__ of exceptions.TypeError object>
    __getattribute__ = <method-wrapper '__getattribute__' of exceptions.TypeError object>
    __getitem__ = <method-wrapper '__getitem__' of exceptions.TypeError object>
    __getslice__ = <method-wrapper '__getslice__' of exceptions.TypeError object>
    __hash__ = <method-wrapper '__hash__' of exceptions.TypeError object>
    __init__ = <method-wrapper '__init__' of exceptions.TypeError object>
    __new__ = <built-in method __new__ of type object>
    __reduce__ = <built-in method __reduce__ of exceptions.TypeError object>
    __reduce_ex__ = <built-in method __reduce_ex__ of exceptions.TypeError object>
    __repr__ = <method-wrapper '__repr__' of exceptions.TypeError object>
    __setattr__ = <method-wrapper '__setattr__' of exceptions.TypeError object>
    __setstate__ = <built-in method __setstate__ of exceptions.TypeError object>
    __sizeof__ = <built-in method __sizeof__ of exceptions.TypeError object>
    __str__ = <method-wrapper '__str__' of exceptions.TypeError object>
    __subclasshook__ = <built-in method __subclasshook__ of type object>
    __unicode__ = <built-in method __unicode__ of exceptions.TypeError object>
    args = ('string indices must be integers, not str',)
    message = 'string indices must be integers, not str'

The above is a description of an error in a Python program.  Here is
the original traceback:

Traceback (most recent call last):
  File "/usr/bin/contrail-api", line 9, in <module>
    load_entry_point('vnc-cfg-api-server==0.1dev', 'console_scripts', 'contrail-api')()
  File "/usr/lib/python2.7/dist-packages/vnc_cfg_api_server/vnc_cfg_api_server.py", line 1572, in server_main
    main()
  File "/usr/lib/python2.7/dist-packages/vnc_cfg_api_server/vnc_cfg_api_server.py", line 1531, in main
    vnc_api_server = VncApiServer(args_str)
  File "/usr/lib/python2.7/dist-packages/vnc_cfg_api_server/vnc_cfg_api_server.py", line 422, in __init__
    self._db_init_entries()
  File "/usr/lib/python2.7/dist-packages/vnc_cfg_api_server/vnc_cfg_api_server.py", line 1074, in _db_init_entries
    config_version=CONFIG_VERSION))
  File "/usr/lib/python2.7/dist-packages/vnc_cfg_api_server/vnc_cfg_api_server.py", line 1125, in _create_singleton_entry
    self._db_conn.dbe_create(obj_type, obj_ids, obj_dict)
  File "/usr/lib/python2.7/dist-packages/vnc_cfg_api_server/vnc_cfg_ifmap.py", line 1701, in wrapper2
    ret = func(self, obj_type, obj_ids, obj_dict)
  File "/usr/lib/python2.7/dist-packages/vnc_cfg_api_server/vnc_cfg_ifmap.py", line 1719, in dbe_create
    (ok, result) = method(obj_ids, obj_dict)
  File "/usr/lib/python2.7/dist-packages/vnc_cfg_api_server/gen/vnc_cassandra_client_gen.py", line 4033, in _cassandra_global_system_config_create
    self._create_prop(bch, obj_ids['uuid'], 'autonomous_system', field)
TypeError: string indices must be integers, not str

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
last test failed because stmonleader failed to boot

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
On last test, one of the node failed to boot with below error while cloning image

GlanceConnectionFailed: Connection to glance failed: Error finding address for https://api.jiocloudservices.com:9292/v2/schemas/image: [Errno 8] _ssl.c:510: EOF occurred in violation of protocol

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
last one failed with the above TypeError. I checked it today and see that this error comes when contrail-api try to do db_init, try to populate the objects to zk, but (some of) the existing objects which fail the operation. I will keep this build (1010) for more in depth root cause analysis.

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by bodepd** [comment]
I see this issue occasionally when I was trying to test out the scaling of cassandra.

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
last test faied with above contrail failures (duplicate zk objects)

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
boot failed for one node :( 2015-04-07 06:04:35.912 17296 ERROR nova.compute.manager [-] Instance failed network setup after 1 attempt(s)
2015-04-07 06:04:35.912 17296 TRACE nova.compute.manager Traceback (most recent call last):
2015-04-07 06:04:35.912 17296 TRACE nova.compute.manager   File "/usr/lib/python2.7/dist-packages/nova/compute/manager.py", line 1540, in _allocate_network_async
2015-04-07 06:04:35.912 17296 TRACE nova.compute.manager     system_metadata=sys_meta)
2015-04-07 06:04:35.912 17296 TRACE nova.compute.manager   File "/usr/lib/python2.7/dist-packages/nova/compute/manager.py", line 628, in _instance_update
2015-04-07 06:04:35.912 17296 TRACE nova.compute.manager     **kwargs)

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
last test filed while attaching fip

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
last test failed because the floating ip was not reachable (no route to host)

**Comment by hkumarmk** [comment]
oh no, i deleted them because of couple of machines got errored

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
last time one node failed to boot, the volume creation failed (the compute log did not have a volume id to check with)

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
last test passed :)

**Comment by bodepd** [comment]
please test this

File: manifests/ceph/osd.pp [hunk 2]
```diff
@@ -156,12 +163,25 @@
...
+    cap          => "mon 'allow *' osd 'allow *' mds 'allow'",
+  }
+
+  ##
+  # Running ceph::key with fake secret with admin just to satisfy condition in ceph module
+  # The condition in ::ceph module may need to be removed, after checking upstream code.
   ##
 
   ::ceph::key { 'admin':
     secret   => 'AQCNhbZTCKXiGhAAWsXesOdPlNnUSoJg7BZvsw==',
```
Comment (bodepd) [inline manifests/ceph/osd.pp]: why is this hard-coded?

**Comment by bodepd** [comment]
one node failed on cp4, from the nova-compute logs on that node:

````
2015-04-07 23:19:40.237 17478 ERROR nova.openstack.common.periodic_task [-] Error during ComputeManager.update_available_resource: Timeout while waiting on RPC response - topic: "<unknown>", RPC method: "<unknown>" info: "<unknown>"
````

**Comment by bodepd** [comment]
please test this

**Comment by bodepd** [comment]
1 node failed with the same error, this time on cp14:

````
2015-04-07 23:34:38.998 7134 TRACE nova.compute.manager [instance: 1af0c98e-8084-4993-b62b-a3a15bc9ba8d] Timeout: Timeout while waiting on RPC response - topic: "<unknown>", RPC method: "<unknown>" info: "<unknown>"
````

**Comment by bodepd** [comment]
please test this

**Comment by bodepd** [comment]
failed b/c of #521

**Comment by bodepd** [comment]
please test this

**Comment by bodepd** [comment]
please test this

**Comment by bodepd** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
last test failed because stmonleader did not come up

**Comment by bodepd** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by bodepd** [comment]
please test this

**Comment by bodepd** [comment]
given the instability of the prod env, we just can't get a green light on jenkins. We have a build missing a compute that otherwise passed (although it took longer than I would like...). Given the critical nature of this patch, I am merging...

**Pull request merged**